contrib/babelfishpg_tsql/src/tsqlUnsupportedFeatureHandler.cpp

```diff
@@ -301,7 +301,11 @@ antlrcpp::Any TsqlUnsupportedFeatureHandlerImpl::visitCreate_or_alter_function(T
 		else if (option->NATIVE_COMPILATION())
 			handle(INSTR_UNSUPPORTED_TSQL_ALTER_FUNCTION_NATIVE_COMPILATION_OPTION, option->NATIVE_COMPILATION());
 		else if (option->execute_as_clause())
-			handle(INSTR_UNSUPPORTED_TSQL_EXECUTE_AS_STMT, "EXECUTE AS", getLineAndPos(option->execute_as_clause()));
+		{
+			auto exec_as = option->execute_as_clause();
+			if (!exec_as->CALLER())
+				handle(INSTR_UNSUPPORTED_TSQL_EXECUTE_AS_STMT, "EXECUTE AS", getLineAndPos(option->execute_as_clause()));
```

Comment:
We should improve the error message by saying which EXECUTE AS is not supported. For example,
`ERROR (Message: 'EXECUTE AS SELF|OWNER|<user>|<login>' is not currently supported in Babelfish)`

Reply:
Fixed in next revision by throwing better error message for execute_as_clause statements in pFUNC/PROC/TRIGGER.